### DISCUSSION TITLE
next/image type declarations are missing on CI

### DISCUSSION THREAD

**Discussion by MendyLanda** [vercel/next.js#52029] [Help] [answered]
### Summary

Importing images works fine locally but fails on CI. 
It complains about the type for the images not found.

The errors:
<img width="1547" alt="image" src="https://github.com/vercel/next.js/assets/54242706/ec0f96a6-24e1-4ee2-aeb7-81a8383bbb80">

How I import my images (just an example, it not the ones in the screenshot, but same issue here)
<img width="658" alt="image" src="https://github.com/vercel/next.js/assets/54242706/4e81ce10-7f55-4b6e-ac55-164501d3d056">

This way it works fine:
<img width="336" alt="image" src="https://github.com/vercel/next.js/assets/54242706/98a56407-8682-4dd0-8b61-d5b1b83e1549">

<details>
  <summary>My folder structure:</summary>

```
apps/
└── nextjs/
    ├── public/
    └── src/
        └── pages/
```

</details>

<details>
  <summary>My main tsconfig:</summary>
  
  ```json
    {
      "compilerOptions": {
        "target": "es2017",
        "lib": ["dom", "dom.iterable", "esnext"],
        "allowJs": true,
        "checkJs": true,
        "skipLibCheck": true,
        "strict": true,
        "forceConsistentCasingInFileNames": true,
        "noEmit": true,
        "esModuleInterop": true,
        "module": "esnext",
        "moduleResolution": "node",
        "resolveJsonModule": true,
        "isolatedModules": true,
        "jsx": "preserve",
        "incremental": true,
        "noUncheckedIndexedAccess": true,
        "noErrorTruncation": true
      },
      "include": [
        ".eslintrc.js",
        "prettier.config.cjs",
        "./packages/config/eslint/index.js"
      ]
    }
  ```
</details>

<details>
  <summary>My nextjs app tsconfig:</summary>

  ```json
    {
      "extends": "../../tsconfig.json",
      "compilerOptions": {
        "baseUrl": ".",
        "paths": {
          "~/*": ["./src/*"]
        },
        "plugins": [{ "name": "next" }]
      },
      "exclude": [],
      "include": [
        "next-env.d.ts",
        "**/*.ts",
        "**/*.tsx",
        "**/*.cjs",
        "**/*.mjs",
        ".next/types/**/*.ts"
      ]
    }
  ```
</details>

<details>
  <summary>My next-env.d.ts file:</summary>

```typescript
  /// <reference types="next" />
  /// <reference types="next/image-types/global" />
  
  // NOTE: This file should not be edited
  // see https://nextjs.org/docs/basic-features/typescript for more information.
```
</details>

<details>
  <summary>next info output</summary>

```
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.16.0
      npm: 9.5.1
      Yarn: 1.22.19
      pnpm: 8.3.1
    Relevant packages:
      next: 13.4.5
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
```

</details>

**Comment by yassh** [comment, marked as answer]
I encountered the same issue in my project. I resolved it simply by committing the `next-env.d.ts` file.

**Reply by ebertmota** [reply]
Thanks man, this helped a lot

**Reply by MendyLanda** [reply]
For anyone using turborepo this will solve the issue:

https://github.com/t3-oss/create-t3-turbo/issues/472#issuecomment-1670063826
